src/crypto/index.js

```diff
@@ -711,8 +767,26 @@ Crypto.prototype.bootstrapSecretStorage = async function({
         const sessionBackupKey = await this.getSecret('m.megolm_backup.v1');
         if (sessionBackupKey) {
             logger.info("Got session backup key from secret storage: caching");
-            const decoded = olmlib.decodeBase64(sessionBackupKey);
-            await this.storeSessionBackupPrivateKey(decoded);
+            // fix up the backup key if it's in the wrong format, and replace
+            // in secret storage
+            const fixedBackupKey = fixBackupKey(sessionBackupKey);
+            if (fixedBackupKey) {
+                await this.storeSecret(
+                    "m.megolm_backup.v1", fixedBackupKey, [newKeyId || oldKeyId],
+                );
+            }
+            const decodedBackupKey = new Uint8Array(olmlib.decodeBase64(
+                fixedBackupKey || sessionBackupKey,
+            ));
+            await this.storeSessionBackupPrivateKey(decodedBackupKey);
+        }
+
+        if (setupNewKeyBackup && !keyBackupInfo) {
```

Comment:
I recently moved this on develop to fix the key backup cache (https://github.com/matrix-org/matrix-js-sdk/pull/1303/commits/060549656e7f4806496daf7113c9e8144f9e868f), and it looks like you have it in the new position several blocks above, so this block can be removed...?

Reply:
Hmm, yes.  `git stash` seems to have done something odd here.